### PR TITLE
Add a Set.prototype.keys() article

### LINKED ISSUE #13286: Issue with "Set": `keys` does not appear as a method in the left sidebar
MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set

#### What information was incorrect, unhelpful, or incomplete?
In the left sidebar, **Methods** includes `Set.prototype.values()` but does not include `Set.prototype.keys()`, presumably because /en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/keys is a redirect to /en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/values and lacks its own content under files/en-us/web/javascript/reference/global_objects/set.

#### Specific section or headline?
See above.

#### What did you expect to see?
I expected to see all methods documented.

#### Did you test this? If so, how?
Yes, and I can get the item to appear by creating files/en-us/web/javascript/reference/global_objects/set/keys/index.md (but this trumps the redirect in every respect, which may not be desirable).

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/javascript/reference/global_objects/set`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/javascript/reference/global_objects/set/index.md
* Last commit: https://github.com/mdn/content/commit/281e3b21178946c8301232a8eb50d11770ee8450
* Document last modified: 2022-02-18T09:53:31.000Z

</details>

### PR DESCRIPTION
**Pull request by sideshowbarker**
Fixes https://github.com/mdn/content/issues/13286

### CODE REVIEW

File: files/en-us/web/javascript/reference/global_objects/set/keys/index.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+{{Specifications("https://tc39.es/ecma262/multipage/keyed-collections.html#sec-set.prototype.keys")}}
+
+## Browser compatibility
+
+{{Compat}}
+
+## See also
+
+- {{jsxref("Set.prototype.entries()")}}
```
Comment (gibson042) [inline files/en-us/web/javascript/reference/global_objects/set/keys/index.md:39]: Should the "See also" for each of `entries()`, `keys()`, and `values()` reference the other two, as is the case for Map methods?

Comment (sideshowbarker) [reply files/en-us/web/javascript/reference/global_objects/set/keys/index.md:40]: Yes, thanks — I’ve pushed an update with that change

**Comment by Josh-Cena** [comment]
I think (also based on the original issue description) there was some infra update needed in Yari? What's the resolution?

**Comment by teoli2003** [comment]
> I think (also based on the original issue description) there was some infra update needed in Yari? What's the resolution?



> I think (also based on the original issue description) there was some infra update needed in Yari? What's the resolution?

No, we just need to have a separate page for the `keys()` method, and it will work. Not having it would need Yari's work, and this would bring extra complexity to the platform to save the creation of a page. This combines with the fact that dev time is sparse to make us choose the _just create the page_ solution.

**Comment by sideshowbarker** [comment]
> I think (also based on the original issue description) there was some infra update needed in Yari? What's the resolution?

The resolution is that we’re not going to do that Yari change after all. Instead, I just updated this article source here to add a `spec_url` key with the https://tc39.es/ecma262/multipage/keyed-collections.html#sec-set.prototype.keys URL.

The result is that we have a Specifications table with two links to the ES spec — one link to the `#ec-set.prototype.keys` section, and one link to the `#ec-set.prototype.values` section.

Given the odd nature of this article, having those two spec-section links in the Specifications table doesn’t seem bad for developers. At least it doesn’t strike me now as being nearly as suboptimal as it seems to have when I originally raised this PR.

**Review by Josh-Cena — APPROVED** [review]
OK—LGTM

(Also worth mentioning it doesn't call `Set.p.values()` but only uses its _initial value_? Not a blocker)

**Pull request merged**